### PR TITLE
add missing public methods to SelectInterface and related.

### PR DESCRIPTION
**Pull request by pavarnos**
make reset() public and add isDistinct() to round out the interface.

### CODE REVIEW

File: src/Common/SelectInterface.php [hunk 1]
```diff
@@ -79,6 +88,35 @@ public function cols(array $cols);
...
+    public function removeCol($alias);
+
+    /**
+     *
+     * Does the query have any columns in it?
+     *
+     * @return bool
+     *
+     */
+    public function hasCols();
```
Comment (harikt) [inline src/Common/SelectInterface.php:118]: I have a problem with this `hasCols` , it sounds like to check if there is a column value exists. But this looks like a check if the columns is empty or not.

**Comment by harikt** [comment]
I will leave this to @pmjones for further review.

File: src/Common/SelectInterface.php (hunk 1 above)
Comment (pavarnos) [reply src/Common/SelectInterface.php:118]: yes. i can see why it was added (so you can check if that part of the query is valid). What about adding `hasCol($name_or_alias)`?

**Comment by pavarnos** [comment]
should we do UpdateInterface and InsertInterface here? The changes are mostly pretty simple (copying the declaration from the real class to the interface declaration). So they should be easy to review as one PR. What do you think?

**Comment by harikt** [comment]
I feel separate may help. The benefit is if something is likely not going to merge can be rejected other PR can get merged.

File: src/Common/SelectInterface.php (hunk 1 above)
Comment (harikt) [reply src/Common/SelectInterface.php:118]: I am ok with that :) .

File: src/Common/Select.php [hunk 2]
```diff
@@ -307,6 +307,20 @@ public function removeCol($alias)
...
+     * Has the column or alias been added to the query?
+     *
+     * @param string $alias The column or alias to look for
+     *
+     * @return bool
+     *
+     */
+    public function hasCol($alias)
+    {
+        return isset($this->cols[$alias]) || array_search($alias, $this->cols) !== false;
```
Comment (harikt) [inline src/Common/Select.php:319]: Why do we need to do a search? Doesn't the isset itself help us ?

Comment (pavarnos) [reply src/Common/Select.php:319]: see https://github.com/auraphp/Aura.SqlQuery/pull/112/files/9fb886d8955663988cdfb5c371416ccdc7cdbe27#diff-1e1715bbfd6b71bb485e70bb72002475R99. cols() can add a column or an alias. This finds both, similar to how removeCol() works.

**Comment by pavarnos** [comment]
this is (hopefully) the last of the changes needed to fill out the *Interface classes. 
I'd originally planned to do this as 2 separate PRs, but found that the changes for the second PR were so small it wasn't worth it So they are all together here.

**Pull request merged**